pydantic/_internal/_std_types_schema.py

```diff
@@ -90,47 +91,32 @@ def to_enum(input_value: Any, /) -> Enum:
-
-    to_enum_validator = core_schema.no_info_plain_validator_function(to_enum)
     if issubclass(enum_type, int):
         # this handles `IntEnum`, and also `Foobar(int, Enum)`
-        updates['type'] = 'integer'
-        lax = core_schema.chain_schema([core_schema.int_schema(), to_enum_validator])
-        # Disallow float from JSON due to strict mode
-        strict = core_schema.json_or_python_schema(
-            json_schema=core_schema.no_info_after_validator_function(to_enum, core_schema.int_schema()),
-            python_schema=strict_python_schema,
-        )
+        js_updates['type'] = 'integer'
+        lax_schema = core_schema.no_info_after_validator_function(to_enum, core_schema.int_schema())
     elif issubclass(enum_type, str):
         # this handles `StrEnum` (3.11 only), and also `Foobar(str, Enum)`
-        updates['type'] = 'string'
-        lax = core_schema.chain_schema([core_schema.str_schema(), to_enum_validator])
-        strict = core_schema.json_or_python_schema(
-            json_schema=core_schema.no_info_after_validator_function(to_enum, core_schema.str_schema()),
-            python_schema=strict_python_schema,
-        )
+        js_updates['type'] = 'string'
+        lax_schema = core_schema.no_info_after_validator_function(to_enum, core_schema.str_schema())
     elif issubclass(enum_type, float):
-        updates['type'] = 'numeric'
-        lax = core_schema.chain_schema([core_schema.float_schema(), to_enum_validator])
-        strict = core_schema.json_or_python_schema(
-            json_schema=core_schema.no_info_after_validator_function(to_enum, core_schema.float_schema()),
-            python_schema=strict_python_schema,
-        )
+        js_updates['type'] = 'numeric'
+        lax_schema = core_schema.no_info_after_validator_function(to_enum, core_schema.float_schema())
     else:
-        lax = to_enum_validator
-        strict = core_schema.json_or_python_schema(json_schema=to_enum_validator, python_schema=strict_python_schema)
+        lax_schema = core_schema.no_info_plain_validator_function(to_enum)
+
     enum_schema = core_schema.lax_or_strict_schema(
-        lax_schema=lax, strict_schema=strict, ref=enum_ref, metadata={'pydantic_js_functions': [get_json_schema]}
+        lax_schema=lax_schema,
+        strict_schema=core_schema.json_or_python_schema(
+            json_schema=lax_schema, python_schema=core_schema.is_instance_schema(enum_type)
+        ),
+        ref=enum_ref,
+        metadata={'pydantic_js_functions': [get_json_schema]},
     )
-    use_enum_values = config.get('use_enum_values', False)
-    if use_enum_values:
-        enum_schema = core_schema.chain_schema(
-            [enum_schema, core_schema.no_info_plain_validator_function(lambda x: x.value)]
-        )
+    if config.get('use_enum_values', False):
+        enum_schema = core_schema.no_info_after_validator_function(attrgetter('value'), enum_schema)
```

Comment:
`attrgetter` is nearly twice as fast as a lambda:

```python
In [1]: import enum

In [2]: class Foo(enum.Enum):
   ...:     x = 'a'
   ...:     y = 'b'
   ...: 

In [3]: with_lambda = lambda thing: thing.x

In [4]: %timeit with_lambda(Foo)
39.1 ns ± 0.161 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [5]: from operator import attrgetter

In [6]: using_attrgetter = attrgetter('x')

In [7]: %timeit using_attrgetter(Foo)
24.5 ns ± 0.473 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

Reply:
Should we be using `attrgetter` more broadly?